app/src/organisms/Desktop/ProtocolDetails/AnnotatedSteps/index.tsx

```diff
@@ -68,31 +68,39 @@ export function AnnotatedSteps(props: AnnotatedStepsProps): JSX.Element {
       <div className={styles.annotated_steps_wrap}>
         {groupedCommandsHighlightedInfo != null &&
         groupedCommandsHighlightedInfo.length > 0
-          ? groupedCommandsHighlightedInfo.map((group, index) =>
-              'annotationIndex' in group ? (
-                <AnnotatedGroup
-                  key={`group_${group.annotationIndex}_${index}`}
-                  analysis={analysis}
-                  annotationType={
-                    annotations[group.annotationIndex]?.machineReadableName
-                  }
-                  subCommands={group.subCommands}
-                  allRunDefs={allRunDefs}
-                  setSelectedCommand={setSelectedCommand}
-                  handlePause={handlePause}
-                />
-              ) : (
-                <IndividualCommand
-                  fromGroup={true}
-                  key={group.command.id}
-                  command={group.command}
-                  isHighlighted={group.isHighlighted}
-                  analysis={analysis}
-                  allRunDefs={allRunDefs}
-                  setSelectedCommand={setSelectedCommand}
-                />
-              )
-            )
+          ? groupedCommandsHighlightedInfo.map((group, index) => {
+              const nextIndex = groupedCommandsHighlightedInfo[index + 1]
+              const nextIsGrouped =
+                nextIndex != null && 'annotationIndex' in nextIndex
+
+              if ('annotationIndex' in group) {
```

Comment:
nit, can remove the `else` block:
```suggestion
              if ('annotationIndex' in group) {
              ...
              }
              return ...
```

Also out of curiosity, what is this mapping doing? I'm not super familiar with command annotations

Reply:
i don't think we can. If your protocol has any command annotations, it will go through this and the commands can be grouped or still individual. this is the `GroupedCommand` type. You can have a LeafNode or ParentNode and the `else` is for LeafNodes

```
interface ParentNode {
  annotationIndex: number
  subCommands: LeafNode[]
  isHighlighted: boolean
}
export interface LeafNode {
  command: RunTimeCommand
  isHighlighted: boolean
}

export type GroupedCommands = Array<LeafNode | ParentNode>
```

Reply:
Ah sorry I meant you can just remove the `else { ... }` wrapper and return the block if the first `if` condition doesn't hit. Not necessary, I just find it a bit cleaner

Reply:
Like 
```
              if ('annotationIndex' in group) {
                return (
                  <AnnotatedGroup
                    key={`group_${group.annotationIndex}_${index}`}
                    analysis={analysis}
                    annotationType={
                      annotations[group.annotationIndex]?.machineReadableName
                    }
                    subCommands={group.subCommands}
                    allRunDefs={allRunDefs}
                    setSelectedCommand={setSelectedCommand}
                    handlePause={handlePause}
                  />
                )
              }
              return (
                <IndividualCommand
                  fromGroup={nextIsGrouped}
                  key={group.command.id}
                  command={group.command}
                  isHighlighted={group.isHighlighted}
                  analysis={analysis}
                  allRunDefs={allRunDefs}
                  setSelectedCommand={setSelectedCommand}
                />
              )
```